packages/manager/modules/manager-pci-common/src/components/rclone-download/StorageRegionSelector.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect, useState } from 'react';
+import { useTranslatedMicroRegions } from '@ovh-ux/manager-react-components';
+import { OdsSelect } from '@ovhcloud/ods-components/react';
+import { TRegion } from '../../api/data';
+
+interface StorageRegionSelectorProps {
+  regions: TRegion[];
+  onRegionChange: (regionName: string) => void;
+}
+
+export function StorageRegionSelector({
+  regions,
+  onRegionChange,
+}: Readonly<StorageRegionSelectorProps>) {
+  const [currentRegion, setCurrentRegion] = useState<string>();
+  const { translateMicroRegion } = useTranslatedMicroRegions();
+
+  // pre-select first region
+  useEffect(() => {
+    if (!currentRegion && regions?.length) {
+      const [region] = regions;
+      setCurrentRegion(region.name);
+      onRegionChange(region.name);
+    }
+  }, [currentRegion, regions, onRegionChange]);
```

Comment:
```suggestion
  }, [regions, onRegionChange]);
```
I don't think you need to put currentRegion as dependency here

Reply:
it is used in the condition inside the useEffect so it seems it should be listed, do you have something else in mind ?

Reply:
You will enter in the useEffect if the currentRegion changed and was not undefined but the useEffect will do nothing